editor/src/components/canvas/utopia-canvas-vars.tsx

```diff
@@ -1,12 +1,20 @@
 import React from 'react'
+import { CanvasScale, CanvasScrollOffset } from '../../utils/global-positions'
 import { useEditorState } from '../editor/store/store-hook'
 
 export const UtopiaCanvasVarStyleTag = React.memo(() => {
   const roundedCanvasOffset = useEditorState(
     (store) => store.editor.canvas.roundedCanvasOffset,
     'DesignPanelRoot roundedCanvasOffset',
   )
+  const canvasScale = useEditorState(
+    (store) => store.editor.canvas.scale,
+    'DesignPanelRoot roundedCanvasOffset',
+  )
 
+  CanvasScrollOffset.x = roundedCanvasOffset.x
+  CanvasScrollOffset.y = roundedCanvasOffset.y
+  CanvasScale.current = canvasScale
```

Comment:
Why is this happening inside a react component? This should be happening as part of the dispatch function or action handling

Reply:
this is deliberate. we are mutating the CanvasScrollOffset variable exactly at the same time we are chaning the value of the css vars in the DOM. this means that code can rely on the two being in sync.

Reply:
Can you add this as a comment please